components/profile/playHistory.tsx

```diff
@@ -0,0 +1,237 @@
+          t._id.toString() === playAttempt._id.toString()
+        ));
+      })
+      );
+    }
+  }, [playHistory]);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (window.innerHeight + window.scrollY >= document.body.offsetHeight - 100) {
+        // Reached bottom
+        if (playHistory && playHistory.length > 0 && !isLoading) {
+          setCursor(playHistory[playHistory.length - 1]._id.toString());
+        }
+      }
+    };
+
+    window.addEventListener('scroll', handleScroll);
+
+    return () => window.removeEventListener('scroll', handleScroll);
+  }, [isLoading, playHistory]);
+
+  const handleSliderChange = (e: any) => {
+    const newValue = e.target.value;
+
+    if (minDurationMinutes === intermediateMinDuration) {
+      return;
+    }
+
+    setIntermediateMinDuration(newValue);
+    setMinDurationMinutes(intermediateMinDuration);
+    setAccumulatedPlayHistory([]); // Clear the play history
+    setCursor(null); // Reset the cursor
+  };
+
+  const display = accumulatedPlayHistory && accumulatedPlayHistory.map((playAttempt, index) => {
+    const level = playAttempt.levelId as EnrichedLevel;
+    let durationInbetween = null;
+
+    if (prevEndTime) {
+      durationInbetween = moment.duration(playAttempt.startTime - prevEndTime, 'seconds').humanize();
+    }
+
+    prevEndTime = playAttempt.endTime;
+
+    const currentDate = moment.unix(playAttempt.startTime).format('MMMM Do, YYYY');
+    const showDate = currentDate !== prevDate;
+
+    prevDate = currentDate;
+
+    const isLeftAligned = index % 2 === 0;
+
+    return (
+      <div
+        key={playAttempt._id.toString()}
+        className='flex flex-col gap-2'
+        style={{ position: 'relative' }}
+      >
+        {showDate && (
+          <div className={'md:text-lg text-xs absolute top-50% transform -translate-y-1/2 text-gray-500 sm:left-[-10%] lg:left-[-50%]'}>
+
+            {currentDate}
+          </div>
+        )}
+
+        { durationInbetween && <span className='text-gray-500 dark:text-gray-400 justify-center self-center'>{durationInbetween} later</span> }
+        <div className={`flex flex-row items-center relative ${isLeftAligned ? 'flex-row-reverse' : ''}`}>
+          <div>
+            <div style={{
+              position: 'absolute',
+              width: '2px',
+              backgroundColor: '#ccc',
+              top: 0,
+              bottom: 0,
+              left: '50%',
+              transform: 'translate(-50%)',
+
+              zIndex: 0,
+            }} />
+            <div style={{
+              position: 'absolute',
+              height: '10px',
+              width: '10px',
+              backgroundColor: '#ccc',
+              borderRadius: '50%',
+              left: '50%',
+              top: '50%',
+              transform: 'translate(-50%, -50%)',
+              zIndex: 1
+            }} />
+          </div>
+          <div className={`${isLeftAligned ? '' : 'mr-10'}`}>
```

Comment:
Noticed some css issues with the alignment here (text overlapping the middle line on low width, text being clipped within the selectcard on low width)

Reply:
I wonder if it would be easier to parse visually + easier to code the css if every row is in this order:

vertical line, level select card, time stats

May be worth trying instead of alternating left/right